### PR TITLE
Add tests for `null` values

### LINKED ISSUE #456: Improve tests for `null` values
Let's say we have two documents in the collection: `{"foo": "bar"}` and `{"bar": null}`. We should check that those queries work in the same way in FerretDB and MongoDB:

- [ ] `{"foo": null}`
- [ ] `{"bar": null}`
- [ ] `{"foo": {"$eq": null}}`
- [x] `{"bar": {"$eq": null}}`

If we already have tests covering that, great. If not, we should add them and fix bugs if they exist.

### PR DESCRIPTION
**Pull request by w84thesun**
Closes #456.

### CODE REVIEW

File: integration/query_comparision_test.go [hunk 1]
```diff
@@ -170,6 +170,29 @@ func TestQueryComparisonEq(t *testing.T) {
 			q:           bson.D{{"value", bson.D{{"$eq", int64(math.MinInt64)}}}},
 			expectedIDs: []any{"int64-min"},
 		},
+		"CompareNoSuchField": {
+			q: bson.D{{"no-such-field", nil}},
```
Comment (AlekSi) [inline integration/query_comparision_test.go:174]: `{"no-such-field": {"$eq": nil}}` is missing.

And please rename test cases and group them by blank lines just like other cases

**Review by AlekSi — APPROVED** [review]
